src/js/components/Button/Button.js

```diff
@@ -264,7 +300,12 @@ const Button = forwardRef(
       }
     };
 
-    let buttonIcon = icon;
+    const hoverIcon = getKindIcon(themePaths?.hover, theme, kind);
+    const kindIcon =
+      hover && hoverIcon
+        ? hoverIcon
+        : getKindIcon(themePaths?.base, theme, kind);
+    let buttonIcon = icon || kindIcon;
```

Comment:
The above getKindIcon() calls are a bit inefficient. What about something like?:
`let buttonIcon = icon || (hover && getKindIcon(themePaths?.hover, theme, kind)) || getKindIcon(themePaths?.base, theme, kind);`

Reply:
Let me look into this, thanks for the suggestion!

Reply:
Simplified this in recent commit to a slightly modified but essentially equivalent logic., Left concept of `kindIcon` for reason detailed in below comment re: snapshot changes on other button-related components.